charm/src/charm.py

```diff
@@ -210,6 +210,11 @@ def fetch_mongodb_relation_data(self) -> dict:
             username = val.get("username")
             password = val.get("password")
             database = val.get("database")
+            break
+        else:
+            logger.error("No database relation data found yet")
+            sys.exit()
```

Comment:
Probably it's a personal preference, but to me this code is kinda hard to read. It took some time to understand what's going on in the loop.

Reply:
Yeah, I had a similar thought about it, but was relucatant to change from the pattern I've seen other charms use. This is looping through relation data that is sent to us by an event. I've only ever personally seen it send one k:v pair when it works, so in theory you could just count the entries and if it's >0 then take the first one.

However I'm not sure why every example I've seen seems to indicate that there *could* be more than one entry. I can change it, but either way, I think it's going to be very hard to debug if anything ever goes wrong with it. Forutnately we should only hit this section very soon after deployment